ports/unix/mpthreadport.c

```diff
@@ -193,6 +196,11 @@ void mp_thread_create(void *(*entry)(void *), void *arg, size_t *stack_size) {
         *stack_size = PTHREAD_STACK_MIN;
     }
 
+    // ensure there is enough stack to include a stack-overflow margin
+    if (*stack_size < 2 * THREAD_STACK_OVERFLOW_MARGIN) {
+        *stack_size = 2 * THREAD_STACK_OVERFLOW_MARGIN;
```

Comment:
This seems fine to me, but I was wondering if RAM is really in short supply if 

    *stack_size += THREAD_STACK_OVERFLOW_MARGIN;

would make more sense or if that is just not a concern.

Reply:
The problem with doing it that way is that it introduces a discontinuity in the amount of stack available: if you do `stack_size(2 * MARGIN - 1)` then it'll allocate `3 * MARGIN - 1`, but if you do `stack_size(2 * MARGIN)` then it'll allocate just `2 * MARGIN`.  That could lead to confusion if one is gradually increasing/decreasing the stack size and testing code.

Reply:
makes sense, thanks